### ISSUE TITLE
D_data_loss and G_discriminator_loss don't change

### ISSUE THREAD

**Issue by axem20** [agrimgupta92/sgan#56]
As in the title, the adversarial losses don't change at all from 1.398 and 0.693 resepectively after roughly epoch 2 until end. Though G_l2_loss does change. Any ideas whats wrong? I've tried changing hyperparameters to those given in the pretrained models as suggested in a previous thread.

**Comment by PhyllisH** [comment]
I met this problem as well. Have u figured out what is wrong?

**Comment by axem20** [comment]
Have not :(

**Comment by PhyllisH** [comment]
You could change the parameter 'l2_loss_weight'. Then the loss would change.

**Comment by axem20** [comment]
You mean reduce the weight of l2_loss? that would encourage the adversarial loss to decrease?

**Comment by PhyllisH** [comment]
I mean that you could change the default value of 'args.l2_loss_weight'.

**Comment by PhyllisH** [comment]
However, the D_data_loss and G_discriminator_loss do not change after several epochs from 1.386 and 0.693 while other losses keep changing.

**Comment by cuihenggang** [comment]
Same question here. My loss doesn't change.

**Comment by JackFram** [comment]
I found out this could be due to the activation function of discriminator is ReLU, and the weight initialization would lead the output be 0 at the beginning, and since ReLU output 0 for all negative value, so gradient is 0 as well. Simply change discriminator's real_classifier's activation function to LeakyReLU could help.

**Comment by ZhoubinXM** [comment]
> l2_loss_weight

change to what

**Comment by ZhoubinXM** [comment]
> I found out this could be due to the activation function of discriminator is ReLU, and the weight initialization would lead the output be 0 at the beginning, and since ReLU output 0 for all negative value, so gradient is 0 as well. Simply change discriminator's real_classifier's activation function to LeakyReLU could help.

I changed activation the loss became Nan

**Comment by Yuliang-Zou** [comment]
Even if I replace ReLU with LeakyReLU, the losses do not change basically.

**Comment by ZhoubinXM** [comment]
> Even if I replace ReLU with LeakyReLU, the losses do not change basically.

U can change the L2_loos_weight. It could be help.

**Comment by Psychic-DL** [comment]
> Even if I replace ReLU with LeakyReLU, the losses do not change basically.
> 
> U can change the L2_loos_weight. It could be help.

When I change the  L2_loos_weight, it appears this error: 
AttributeError: 'dict' object has no attribute 'l2_loss'.
It in generator_step()
![image](https://user-images.githubusercontent.com/30741409/69605951-7e497380-105c-11ea-84e8-2c007e9f1b79.png)

Could you please help me find out the reason? Thank you.

**Comment by zpp960807** [comment]
I have met the same problem,even if I set the l2_liss_weight to 1, the adversarial losses didn't change yet and it was still 1.386 and 0.693.

**Comment by buzhanpeng** [comment]
If running on the Windows operating system, all parameters should be based on the run_ traj. The sh file is modified. You cannot run the train.py program directly

**Comment by buzhanpeng** [comment]
If running on the Windows operating system, all parameters should be based on the run_ traj. sh. You cannot run the train.py program directly

**Comment by neugzy** [comment]
> I found out this could be due to the activation function of discriminator is ReLU, and the weight initialization would lead the output be 0 at the beginning, and since ReLU output 0 for all negative value, so gradient is 0 as well. Simply change discriminator's real_classifier's activation function to LeakyReLU could help.

Yes, using LeakyReLU could help change the D_loss and G_loss. But it seems that the evaluate results of relu and leakyRelu make no difference. Both of them can get reasonable results.

**Comment by sjx3906** [comment]
> > I found out this could be due to the activation function of discriminator is ReLU, and the weight initialization would lead the output be 0 at the beginning, and since ReLU output 0 for all negative value, so gradient is 0 as well. Simply change discriminator's real_classifier's activation function to LeakyReLU could help.
> 
> Yes, using LeakyReLU could help change the D_loss and G_loss. But it seems that the evaluate results of relu and leakyRelu make no difference. Both of them can get reasonable results.

I changed relu to leakyrelu both in the generator and the discriminator, the loss did chenged, but it's very strange, Is this normal?
![image](https://github.com/agrimgupta92/sgan/assets/110144586/5e591151-07b0-49b9-95b1-00f076d453df)
